### DISCUSSION TITLE
[UI Update] The Control Type Filter is finally implemented

### DISCUSSION THREAD

**Discussion by lllyasviel** [Mikubill/sd-webui-controlnet#1423] [General]
Beginning from 1.1.192, we will add a quick shortcut in UI, the Control Type filter.

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/3b258400-25a7-414c-9fa9-2bc0b4cd9961)

You can turn off this quick filter in setting if you prefer not to have it, using the option “Disable control type selection”:

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/6ebbecd5-d5fe-446a-8a6d-48752153d86f)

This filter has two functionalities:

First, if you select one type of control, your preprocessor list and model list will only show items of that type. You can always select “All” to see all preprocessors/models just like before. For example, I select “depth”

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/a5c25cbd-2c0b-4c51-92b4-f865d67c4738)

Then

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/e67c245a-e681-4224-a072-1446b84998d5)

Second, if you select one type of control, it will automatically select a default combination of that control. For example, I select “depth”

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/e619ba74-f4a2-4bae-9a18-28ec3221a6c3)

Then

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/b61e6e11-5edf-4595-9c84-03ebda4cc9f2)

Note that this filter has nothing to do with API and code logics so it will lead to zero influence to all other extensions.

Now we can swift methods much easier. And we do not need to worry about the list growing longer and longer.

**You need at least 1.1.192 to use it.**

**Comment by lllyasviel** [comment]
(Now we can use/shift simple controls like canny/depth in one click - finally - this will really save our life time)

**Comment by lllyasviel** [comment]
(Spoiler: the next update will probably bring a new tile preprocessor, and since now we have this quick shortcut filter, we do not need to worry any more about the list growing too much longer)

**Comment by djbielejeski** [comment]
@lllyasviel Do you have plans to allow us to save presets within the extension?  Example: Would like to setup 3 control nets, 1st Canny with a specific weight, 2nd Depth with a specific weight, and 3rd with Pose.   Would be nice to one click restore those settings
